src/windows/mod.rs

```diff
@@ -233,7 +233,13 @@ pub const SIGSEGV: ::c_int = 11;
 pub const SIGTERM: ::c_int = 15;
 pub const SIGABRT: ::c_int = 22;
 pub const NSIG: ::c_int = 23;
+
 pub const SIG_ERR: ::c_int = -1;
```

Comment:
`#define SIG_ERR ((_crt_signal_t)-1)`

Also `libc` defines `sighandler_t` as `usize` when in reality it is:
`typedef void (__CRTDECL* _crt_signal_t)(int);`

Reply:
> Also libc defines sighandler_t as usize when in reality it is:
typedef void (__CRTDECL* _crt_signal_t)(int);

I'll take a PR to fix this, but I'll prefer if this change were in a different PR, and not here. We probably need to do a crater run while rolling in this change.

Reply:
Just keep in mind we can't naively use `extern "C" fn(c_int)` as the type because  `SIG_DFL` has a value of 0.

Reply:
Yes, we need `Option<extern "C" fn(c_int) -> ()>` or similar (I'm not sure what to do about the `__CRTDECL` or whether `extern "system"` or some other ABI has to be used for windows.

Reply:
`__CRTDECL` is `__cdecl` which is `extern "cdecl"` or `extern "C"`. Most of the CRT uses `__CRTDECL`.